src/llm-eval/README.md

```diff
@@ -0,0 +1,297 @@
+If not, then follow these steps:
+
+1. Install .NET 8 or higher
+2. Install Visual Studio 2022 or Visual Studio Code
+
+## Models to test, to evaluate and to generate data
+
+It's best to use a GPT-4 model for performing the evaluation, even if your chat app uses GPT-3.5 or another model. You can either use an Azure OpenAI instance, an openai.com instance or any LLM supported by Semantic Kernel.
+
+The KernelFactory class `src/LLMEval.Test/KernelFactory.cs`, creates the necessary kernels for evaluation, generating ground data and the LLM to be tested.
+
+The following code, shows how to create 3 different kernels:
+ 
+```csharp
+// ========================================
+// create kernels
+// ========================================
+var kernelEval = KernelFactory.CreateKernelEval();
+var kernelTest = KernelFactory.CreateKernelEval();
+var kernelGenData = KernelFactory.CreateKernelGenerateData();
+```
+
+### Creating Kernels for Azure OpenAI Services or OpenAI APIs
+
+You can use .NET User Secrets to store the necessary information to create the kernels. In example run these commands in bash to store the information related to an Azure OpenAI Service Endpoint:
+
+```bash
+dotnet user-secrets init
+dotnet user-secrets set "AZURE_OPENAI_MODEL" "Azure OpenAI Deployment Name"
+dotnet user-secrets set "AZURE_OPENAI_ENDPOINT" "Azure OpenAI Deployment Endpoint"
+dotnet user-secrets set "AZURE_OPENAI_KEY" "Azure OpenAI Deployment Key"
+```
+
+Now the KernelFactory class can create a kernel with the following code:
+
+```csharp
+public static Kernel CreateKernelEval()
+{
+    var config = new ConfigurationBuilder().AddUserSecrets<Program>().Build();
+
+    var builder = Kernel.CreateBuilder();
+
+    builder.AddAzureOpenAIChatCompletion(
+        config["AZURE_OPENAI_MODEL"],
+        config["AZURE_OPENAI_ENDPOINT"],
+        config["AZURE_OPENAI_KEY"]);
+
+    return builder.Build();
+}
+```
+
+To learn more about Project Secrets in Development * [Project Secrets in Development](#https://learn.microsoft.com/en-us/aspnet/core/security/app-secrets?view=aspnetcore-8.0&tabs=windows)
+
+### Creating Kernels to access Open Source LLMs, like Llama 3 or Phi-3
+
+Semantic Kernel also allows the access to any model that supports the [OpenAI Chat Completions API](https://github.com/ollama/ollama/blob/main/docs/openai.md). 
+
+The following example creates a kernel using a Llama 3 model deployed and hosted locally using ollama. 
+
+```csharp
+public static Kernel CreateKernelEval()
+{
+    var config = new ConfigurationBuilder().AddUserSecrets<Program>().Build();
+
+    var builder = Kernel.CreateBuilder();
+
+        builder.AddOpenAIChatCompletion(
+            modelId: "llama3",
+            endpoint: new Uri("http://localhost:11434"),
+            apiKey: "api");
+
+    return builder.Build();
+}
+```
+
+This blog explains how to host a local LLM using ollama, [Llama 3](https://ollama.com/blog/llama3)
```

Comment:
Consider replacing the link here to the local models sample.

Reply:
Done!